### PR TITLE
create multiple BUILD files in vendor.

### PR DESCRIPTION
**Pull request by mikedanese**
this should reduce the rebase misery that is vendor/BUILD.

### CODE REVIEW

**Comment by liggitt** [comment]
what is MultiBUILDVendor?

**Comment by spxtr** [comment]
gazel change is merged, should we go with this?

**Comment by ixdy** [comment]
can you make it more obvious which of the commits is autogenerated?

**Comment by ixdy** [comment]
@mikedanese want to tag a new gazel release and rebase this?

**Comment by mikedanese** [comment]
I'll regenerate this today.

**Comment by spxtr** [comment]
Plan on doing this soon?

**Comment by spxtr** [comment]
@k8s-bot cross build this
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by ixdy** [comment]
should we rebase this and try to get it in?

we probably need to find a top-level approver to make this easier.

**Comment by ixdy** [comment]
@bgrant0607 @brendandburns @dchen1107 @jbeda @lavalamp @smarterclayton @thockin @wojtek-t

This change is breaking up the enormous `BUILD` file in `vendor/` into many individual `BUILD` files under `vendor/`, which should reduce merge conflicts significantly. It affects nearly every `BUILD` file in the repo, however, so we need a top-level approver to make this process easier.

**Comment by smarterclayton** [comment]
I'm supportive of this and willing to help approve to make it happen.

**Comment by smarterclayton** [comment]
Let's coordinate tomorrow or Thursday to land this - just ping me on slack

**Comment by mikedanese** [comment]
@k8s-bot node e2e test this

**Comment by ixdy** [comment]
/lgtm

I didn't look at the "autogenerated" commit - I assume it contains only autogenerated changes.

File: .gazelcfg.json [hunk 1]
```diff
@@ -10,7 +10,10 @@
 		"./examples"
 	],
 	"SkippedPaths": [
-		"^_.*"
+		"^_.*",
+		"^staging.*",
+		"_vendor"
```
Comment (ixdy) [inline .gazelcfg.json]: what is `_vendor`?

File: staging/copy.sh [hunk 2]
```diff
@@ -217,7 +217,7 @@ if [ "${FAIL_ON_CHANGES}" = true ]; then
   echo "running FAIL_ON_CHANGES"
   # ignore base.go in diff
   cp "${CLIENT_REPO}/pkg/version/base.go" "${CLIENT_REPO_TEMP}/pkg/version/"
-  if diff -NauprB  -I '^\s*\"Comment\"' -I "GoVersion.*\|GodepVersion.*" "${CLIENT_REPO}" "${CLIENT_REPO_TEMP}"; then
+  if diff -NauprB -I 'BUILD' -I '^\s*\"Comment\"' -I "GoVersion.*\|GodepVersion.*" "${CLIENT_REPO}" "${CLIENT_REPO_TEMP}"; then
```
Comment (ixdy) [inline staging/copy.sh]: might be worth commenting why we're ignoring BUILD files?

File: .gazelcfg.json (hunk 1 above)
Comment (mikedanese) [inline .gazelcfg.json]: @ixdy they are vendors inside of staging.

**Comment by spxtr** [comment]
Do we just keep retrying, or should we ask build-cop to just click the merge button after a rebase?

**Comment by ixdy** [comment]
I'm kinda curious why it didn't merge overnight.

**Comment by smarterclayton** [comment]
Merge, damn you

**Comment by ixdy** [comment]
@k8s-bot pull-kubernetes-cross test this

**Comment by ixdy** [comment]
oh, hm, that actually looks broken:
```
I0414 11:07:29.245] # k8s.io/kubernetes/pkg/util/term
I0414 11:07:29.245] pkg/util/term/resizeevents_windows.go:28: undefined: Size
I0414 11:07:29.245] pkg/util/term/setsize_unsupported.go:21: undefined: Size
```

but not due to this PR.

**Comment by spxtr** [comment]
I think we should just ask build-cop to click the merge button after a rebase.

**Comment by smarterclayton** [comment]
I agree - this is important and continually bumped by everything.

**Comment by mikedanese** [comment]
Flake.

@k8s-bot cvm gce e2e test this

**Pull request merged**